selfdrive/ui/sunnypilot/qt/offroad/settings/vehicle_panel.cc

```diff
@@ -35,6 +70,82 @@ void VehiclePanel::showEvent(QShowEvent *event) {
 
 void VehiclePanel::updatePanel(bool _offroad) {
   platformSelector->refresh(_offroad);
-
   offroad = _offroad;
 }
+
+void VehiclePanel::updateToggles(bool offroad_transition) {
+  updatePanel(offroad_transition);
+  updateCarToggles();
+}
+
+VehiclePanel::ToggleState VehiclePanel::getToggleState(bool hasOpenpilotLong) const {
+  if (!hasOpenpilotLong) {
+    return ToggleState::DISABLED_LONGITUDINAL;
+  }
+  if (!uiState()->scene.started) {
+    return ToggleState::ENABLED;
+  }
+  return ToggleState::DISABLED_DRIVING;
+}
+
+void VehiclePanel::updateToggleState(AbstractControlSP* toggle, bool hasOpenpilotLong) {
+  static const QString LONGITUDINAL_MSG = tr("Enable openpilot longitudinal control first.");
+  static const QString DRIVING_MSG = tr("Cannot modify while driving. Please go offroad mode first.");
+
+  ToggleState state = getToggleState(hasOpenpilotLong);
+
+  switch (state) {
+    case ToggleState::ENABLED:
+      toggle->setDescription(toggle->property("originalDesc").toString());
+      break;
+    case ToggleState::DISABLED_LONGITUDINAL: {
+      QString msg = "<font color='orange'>" + tr("Enable openpilot longitudinal control first to modify this setting.") + "</font>";
+      toggle->setDescription(msg);
+      toggle->showDescription();
+      break;
+    }
+    case ToggleState::DISABLED_DRIVING: {
+      QString msg = "<font color='orange'>" + tr("Cannot modify while driving. Please go offroad mode first.") + "</font>";
+      toggle->setDescription(msg);
+      toggle->showDescription();
+      break;
+    }
+  }
```

Comment:
Maybe instead of the toggle state I'd do something more like an early exit if not OP long 

```suggestion

    if(!hasOpenpilotLong) {
      QString msg = "<font color='orange'>" + tr("Enable openpilot longitudinal control first to modify this setting.") + "</font>";
      toggle->setDescription(msg);
      toggle->showDescription();
      return;
    }
    
  switch (state) {
    case ToggleState::ENABLED:
      toggle->setDescription(toggle->property("originalDesc").toString());
      break;
    case ToggleState::DISABLED_DRIVING: {
      QString msg = "<font color='orange'>" + tr("Cannot modify while driving. Please go offroad mode first.") + "</font>";
      toggle->setDescription(msg);
      toggle->showDescription();
      break;
    }
  }
```

Reply:
Nuked it in favor of more concise language